### PR TITLE
Component unification (Edit profile page)

### PR DESCRIPTION
**Pull request by prachigarg19**
Unify mui components in the webapp.

### CODE REVIEW

**Review by norbertschuler — APPROVED** [review]

**Comment by prachigarg19** [comment]
The layout breaks for empty inputs:
image

**Comment by prachigarg19** [comment]
> The layout breaks for empty inputs: image

Dealt with in latest commit.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 1]
```diff
@@ -198,7 +219,7 @@ export default function EditProfile({}: Props) {
     onDropAccepted: () => {},
   });
 
-  const saveProfile = async (data: any) => {
+  const saveProfile = async (data: object) => {
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:222]: Avoid using `object` or similar generic types. In this case, you could define the type of data submitted by the form.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 2]
```diff
@@ -287,32 +307,40 @@ export default function EditProfile({}: Props) {
...
+            onChange={(event, newType) => {
+              if (newType) {
+                setAccountType(newType.value);
+                setLocalType(newType);
+              }
+            }}
+            renderInput={(params) => (
+              <TextField {...params} label={t('editProfile:iamA')} />
+            )}
+          />
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:334]: Resolve the TS warnings seen here.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 3]
```diff
@@ -337,26 +365,27 @@ export default function EditProfile({}: Props) {
...
-            <MaterialTextField
+            <TextField
               label={t('donate:email')}
               variant="outlined"
               name="email"
               defaultValue={user.email}
-              disabled={true}
-            />
+              inputProps={{ readOnly: true }}
+              disabled
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:374]: are both of these needed?

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:1]: I see a few warnings in the browser console while running this locally. Try to resolve these. 

```
MUI: The value provided to Autocomplete is invalid. None of the options match with `{"id":"","title":"","value":""}`.
```
```
getNews is missing in the 'defaultValue' prop of either its Controller (https://react-hook-form.com/api#Controller) or useForm (https://react-hook-form.com/api#useForm) 
```
```
isPrivate is missing in the 'defaultValue' prop of either its Controller (https://react-hook-form.com/api#Controller) or useForm (https://react-hook-form.com/api#useForm) 
```

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 4]
```diff
@@ -287,32 +307,40 @@ export default function EditProfile({}: Props) {
...
+              if (newType) {
+                setAccountType(newType.value);
+                setLocalType(newType);
+              }
+            }}
+            renderInput={(params) => (
+              <TextField {...params} label={t('editProfile:iamA')} />
+            )}
+          />
         ) : null}
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:335]: You could also use the `<ReactHookFormSelect>` component that is used in BankDetailsForm and PayoutScheduleForm.

**Review by mohitb35 — CHANGES_REQUESTED** [review]
@prachigarg19 Please go through my feedback.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 5]
```diff
@@ -287,32 +307,40 @@ export default function EditProfile({}: Props) {
...
-            select
-            defaultValue={type}
-          >
-            {profileTypes.map((option) => (
-              <MenuItem
-                key={option.value}
-                value={option.value}
-                onClick={() => setAccountType(option.value)}
-              >
+          <MuiAutoComplete
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:312]: You could also use <ReactHookFormSelect> over here, as used in BankDetailsForm / PayoutScheduleForm, if you want to make this available to ReactHookForm.

File: src/features/user/Settings/EditProfile/index.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import { ReactElement } from 'react';
+import CenteredContainer from '../../../common/Layout/CenteredContainer';
+import DashboardView from '../../../common/Layout/DashboardView';
+import EditProfileForm from './EditProfileForm';
+
+export default function (): ReactElement | null {
+  const { t, ready } = useTranslation('me');
+
+  return ready ? (
+    <DashboardView title={t('editProfile')} subtitle={null} type="other">
```
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/features/user/Settings/EditProfile/index.tsx]: Created a separate issue #1758 for later

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 7]
```diff
@@ -298,12 +360,12 @@ export default function EditProfile({}: Props) {
           </div>
           <div style={{ width: '20px' }}></div>
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:361]: Is this needed any more?

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 8]
```diff
@@ -1,37 +1,56 @@
...
+  type?: string;
+};
+
+export default function EditProfileForm() {
   const [snackbarOpen, setSnackbarOpen] = useState(false);
-  const { setErrors } = React.useContext(ErrorHandlingContext);
-  const { user, setUser, token, contextLoaded, logoutUser } =
+  const { handleError } = React.useContext(ErrorHandlingContext);
+  const { user, setUser, token, contextLoaded, impersonatedEmail } =
     React.useContext(UserPropsContext);
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:54]: You seem to have reintroduced some deleted code while merging conflicts.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 9]
```diff
@@ -447,7 +509,8 @@ export default function EditProfile({}: Props) {
             name="isPrivate"
             control={control}
             inputRef={register()}
-            render={(props: any) => (
+            defaultValue={false}
+            render={(props: object) => (
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:513]: There are TS warnings seen here as `props` has been typed as an object

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx
[the remark this answers is not in the document]
Comment (mohitb35) [reply src/features/user/Settings/EditProfile/EditProfileForm.tsx:1]: I still see some warnings relating to the code added that could be resolved.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx [hunk 10]
```diff
@@ -1,37 +1,56 @@
...
+  bio: string;
+  city: string;
+  firstname: string;
+  getNews: boolean;
+  isPrivate: boolean;
+  lastname: string;
+  name: string;
+  url: string;
+  zipCode: string;
+  type?: string;
```
Comment (mohitb35) [inline src/features/user/Settings/EditProfile/EditProfileForm.tsx:47]: `type` should not be included within `FormData` as the field providing this is not registered with react hook form. In `saveProfile` we get the value for `type` from state, not from form data.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx (hunk 2 above)
Comment (mohitb35) [reply src/features/user/Settings/EditProfile/EditProfileForm.tsx]: This is pending

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx (hunk 4 above)
Comment (mohitb35) [reply src/features/user/Settings/EditProfile/EditProfileForm.tsx:346]: Left as is as this needs a greater refactor to the code of the `ReactHookFormSelect` component to make it more generic.

File: src/features/user/Settings/EditProfile/EditProfileForm.tsx (hunk 5 above)
Comment (mohitb35) [reply src/features/user/Settings/EditProfile/EditProfileForm.tsx:323]: Left as is as this needs a greater refactor to the code of the `ReactHookFormSelect` component to make it more generic.

**Review by mohitb35 — APPROVED** [review]
LGTM

**Pull request merged**